purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesOrchestrator.kt

```diff
@@ -900,6 +907,39 @@ internal class PurchasesOrchestrator(
     }
 
     // endregion
+
+    /**
+     * Fetches fresh offerings with rate limiting to prevent excessive network requests.
+     *
+     * @param callback Callback to handle the result
+     * @return true if fresh fetch was triggered, false if rate limited
+     */
+    internal fun fetchOfferingsWithRateLimit(callback: (Offerings?, PurchasesError?) -> Unit): Boolean {
+        return if (preferredLocaleOverrideRateLimiter.shouldProceed()) {
+            log(LogIntent.DEBUG) { "Fetching fresh offerings" }
+            getOfferings(
```

Comment:
A couple possible edge case of this... If our backend is down (very unlikely), or this takes a bit longer for any reason, the paywall might be displayed with the cached offerings, which could be in the wrong locale... Not sure how big of a deal this is it though...

We could clear the offerings cache before fetching it to at least make sure we don't use the wrong locale... though I guess in that case we might be delaying displaying the paywall, or might cause some errors if the request fails for any reason...